registry/storage/driver/gcs/gcs.go

```diff
@@ -369,18 +379,15 @@ type writer struct {
 	sessionURI string
 	buffer     []byte
 	buffSize   int
+	GCS        *storage.Client
 }
 
 // Cancel removes any written content from this FileWriter.
 func (w *writer) Cancel() error {
 	w.closed = true
-	err := storageDeleteObject(cloud.NewContext(dummyProjectID, w.client), w.bucket, w.name)
-	if err != nil {
-		if status, ok := err.(*googleapi.Error); ok {
-			if status.Code == http.StatusNotFound {
-				err = nil
-			}
-		}
+	err := storageDeleteObject(context.Background(), w.bucket, w.name, w.GCS)
+	if err == storage.ErrObjectNotExist {
+		return storagedriver.PathNotFoundError{Path: w.name}
```

Comment:
It looks like before the object would not error if code was not found. Is this a new intended behaviour?

Reply:
This is probably a mistake on adapting things on my part. Thank you.